torchtext/vocab.py

```diff
@@ -40,21 +40,38 @@ def __init__(self, counter, max_size=None, min_freq=1, specials=['<pad>'],
                 vocabulary. Values less than 1 will be set to 1. Default: 1.
             specials: The list of special tokens (e.g., padding or eos) that
                 will be prepended to the vocabulary in addition to an <unk>
-                token. Default: ['<pad>']
+                token. Default: ['<pad>'].
+            is_label: remove from the list the <unk> token not needed for 
+                the label list
             vectors: one of either the available pretrained vectors
                 or custom pretrained vectors (see Vocab.load_vectors);
                 or a list of aforementioned vectors
         """
         self.freqs = counter.copy()
         min_freq = max(min_freq, 1)
-        counter.update(['<unk>'] + specials)
+        if is_label==False:
+            counter.update(['<unk>'] + specials)
 
-        self.stoi = defaultdict(_default_unk_index)
-        self.stoi.update({tok: i for i, tok in
+            self.stoi = defaultdict(_default_unk_index)
+        
+            self.stoi.update({tok: i for i, tok in
                           enumerate(['<unk>'] + specials)})
-        self.itos = ['<unk>'] + specials
+            self.itos = ['<unk>'] + specials
 
-        counter.subtract({tok: counter[tok] for tok in ['<unk>'] + specials})
+            counter.subtract({tok: counter[tok] for tok in ['<unk>'] + specials})
+        
+        else:
+            counter.update(specials)
```

Comment:
as it currently stands, this would add the `<pad>` token to your class list as well.

Reply:
Okay in that case, with the is_label option I can also remove the "specials"